rskj-core/src/main/java/co/rsk/peg/BridgeSupport.java

```diff
@@ -1017,13 +1017,51 @@ private void logUpdateCollections(Transaction rskTx) {
         eventLogger.logUpdateCollections(sender);
     }
 
-    protected void processSVPFailure(Federation proposedFederation) {
+    protected void updateSvpState(Transaction rskTx) {
+        Optional<Federation> proposedFederationOpt = federationSupport.getProposedFederation();
+        if (proposedFederationOpt.isEmpty()) {
+            logger.debug("[updateSvpState] Proposed federation does not exist, so there's no svp going on.");
+            return;
+        }
+
+        // if the proposed federation exists and the validation period ended,
+        // we can conclude that the svp failed
+        Federation proposedFederation = proposedFederationOpt.get();
+        if (!validationPeriodIsOngoing(proposedFederation)) {
+            logger.info(
+                "[updateSvpState] Proposed federation validation failed at block {}. SVP failure will be processed",
+                rskExecutionBlock.getNumber()
+            );
+            processSVPFailure(proposedFederation);
+            return;
+        }
+
+        Keccak256 rskTxHash = rskTx.getHash();
+        // if the fund tx signed is present, then the fund transaction change was registered,
+        // meaning we can create the spend tx.
+        Optional<BtcTransaction> svpFundTxSigned = provider.getSvpFundTxSigned();
+        if (svpFundTxSigned.isPresent()) {
```

Comment:
Should we also be checking that there is no spend tx created? Just in case.

I know there should only be one at a time so if both are found perhaps we have a problem